lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts

```diff
@@ -549,13 +550,136 @@ describe("Acquires a token successfully via an IMDS Managed Identity", () => {
             );
         });
 
-        test("ignores a cached token when claims are provided", async () => {
+        test('checks if a token was cached or not and if was "bypass_cache=true" was sent to ESTS depending on whether or not claims are provided, and ensures "xms=client-capabilites-string" was sent to ESTS when client capabilities are provided via the Managed Identity Request Parameters', async () => {
```

Comment:
I think you should also include tests when claims are passed, and client capabilities are not and when client capabilities are passed, and claims are not.

Reply:
I think most are covered, but the tests are somewhat difficult to read. 

There are 3 dimentions for this test, each with 2 possible values. So a total of 2^3 = 8 combos.

CP1: {sent, not_sent}
Claims: {sent, not_sent}
Token_in_cache? { yes, no}